packages/swap/src/index.ts

```diff
@@ -538,14 +538,15 @@ export class Swap {
     // Decode the return value.
     //
     // TODO: Expose the event parsing api in anchor to make this less manual.
-    let didSwapEvent = resp.value.logs
+    const didSwapEvent = resp.value.logs
       .filter((log) => log.startsWith('Program log: 4ZfIrPLY4R'))
       .map((log) => {
         const logStr = log.slice('Program log: '.length);
         const logArr = Buffer.from(base64.toByteArray(logStr));
-        return this.program.coder.events.decode('DidSwap', logArr.slice(8));
+        return this.program.coder.events.decode('DidSwap' + logArr.slice(8));
```

Comment:
This is not correct.

Reply:
After the update, `events.decode()` only accepted a single parameter. Any thoughts on how to proceed?

Reply:
Pass in the entire buffer without slicing.

Reply:
is there any other way around